bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/WebexTeamsHandler.java

```diff
@@ -0,0 +1,428 @@
+            return null;
+        }
+    }
+
+    public boolean equalsThingUID(String thingUID) {
+        return getThing().getUID().getAsString().equals(thingUID);
+    }
+
+    public String formatAuthorizationUrl(String redirectUri) {
+        try {
+            if (this.configured) {
+                return this.authService.getAuthorizationUrl(redirectUri, null, thing.getUID().getAsString());
+            } else {
+                return "";
+            }
+        } catch (final OAuthException e) {
+            logger.debug("Error constructing AuthorizationUrl: ", e);
+            return "";
+        }
+    }
+
+    // mainly used to refresh the auth token when using OAuth
+    private boolean refresh() {
+        synchronized (refreshSynchronization) {
+            Person person;
+            try {
+                person = client.getPerson();
+                String type = person.getType();
+                if (type == null) {
+                    type = "?";
+                }
+                updateProperty(PROPERTY_WEBEX_TYPE, type);
+                this.accountType = type;
+                updateProperty(PROPERTY_WEBEX_NAME, person.getDisplayName());
+
+                // Only when the identity is a person:
+                if ("person".equalsIgnoreCase(person.getType())) {
+                    String status = person.getStatus();
+                    updateState(CHANNEL_STATUS, StringType.valueOf(status));
+                    DateFormat df = new SimpleDateFormat(ISO8601_FORMAT);
+                    String lastActivity = df.format(person.getLastActivity());
+                    updateState(CHANNEL_LASTACTIVITY, new DateTimeType(lastActivity));
+                }
+                updateStatus(ThingStatus.ONLINE);
+                return true;
+            } catch (WebexTeamsException e) {
+                logger.warn("Failed to refresh: {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+            return false;
+        }
+    }
+
+    private void startRefresh() {
+        synchronized (refreshSynchronization) {
+            if (refresh()) {
+                cancelSchedulers();
+                if (active) {
+                    refreshFuture = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refreshPeriod,
+                            TimeUnit.SECONDS);
+                }
+            }
+        }
+    }
+
+    /**
+     * Cancels all running schedulers.
+     */
+    private synchronized void cancelSchedulers() {
+        if (refreshFuture != null) {
+            refreshFuture.cancel(true);
+        }
+    }
+
+    public String getUser() {
+        return thing.getProperties().getOrDefault(PROPERTY_WEBEX_NAME, "");
+    }
+
+    public ThingUID getUID() {
+        return thing.getUID();
+    }
+
+    public String getLabel() {
+        return thing.getLabel() == null ? "" : thing.getLabel().toString();
```

Comment:
Not sure ".toString()" is required as `thing.getLabel()` probably returns a String.

Reply:
Updated.